### PR TITLE
Optimize getGlamorClassName

### PR DESCRIPTION
**Pull request by tkh44**
By shaping the string class names beginning with 'css-' we can
just give the resulting object to glamor. This offloads almost the
entire workload to glamor from glamorous. Another benefit is that
now there will be at most one `css-[hash]` class name on an element

The only failing test after the modifications was an element that had 2 `css` class names only having one now. The snapshot was updated to reflect this bug fix.

**Checklist**:

- [ ] Documentation N/A
- [x] Tests 
- [x] Ready to be merged 
- [x] Added myself to contributors table

### CODE REVIEW

File: src/__tests__/__snapshots__/index.js.snap [hunk 1]
```diff
@@ -101,13 +101,13 @@ exports[`can use pre-built glamorous components with css prop 1`] = `
...
 exports[`css prop with a className 1`] = `
-.glamor-1,
-[data-glamor-1] {
+.glamor-0,
+[data-glamor-0] {
   color: red;
 }
 
 <section
-  class="glamor-0 glamor-1 abc-123"
```
Comment (kentcdodds) [inline src/__tests__/__snapshots__/index.js.snap]: :tada: I hadn't even noticed this bug! Might actually help me with something for work too! Woo!

File: src/get-glamor-classname.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import {css, styleSheet} from 'glamor'
+import {css} from 'glamor'
```
Comment (kentcdodds) [inline src/get-glamor-classname.js]: I _love_ not having to reach into the glamor to get the `styleSheet`. This is awesome! :clap:

File: src/get-glamor-classname.js [hunk 3]
```diff
@@ -76,21 +86,3 @@ function handleStyles(styles, props, theme, context) {
   }
   return {mappedArgs, nonGlamorClassNames}
 }
-
-function processStringClass(str, mappedArgs, nonGlamorClassNames) {
```
Comment (kentcdodds) [inline src/get-glamor-classname.js]: :clap: for code removal

File: src/get-glamor-classname.js [hunk 4]
```diff
@@ -10,17 +10,23 @@ import {css, styleSheet} from 'glamor'
...
+      glamorlessClassName.push(name)
     }
-    return groups
-  }, {glamorlessClassName: '', glamorStyles: []})
+  })
+
+  return {glamorlessClassName, glamorStyles}
+}
+
+function buildGlamorSrcFromClassName(className) {
```
Comment (kentcdodds) [inline src/get-glamor-classname.js]: Could we add a comment above this line giving a brief explanation of how this black magic works?

File: src/get-glamor-classname.js [hunk 5]
```diff
@@ -33,18 +39,14 @@ function getGlamorClassName({
   theme,
   context,
 }) {
-  const {
-    glamorStyles: parentGlamorStyles,
-    glamorlessClassName,
-  } = extractGlamorStyles(props.className)
   const {mappedArgs, nonGlamorClassNames} = handleStyles(
-    [...styles, parentGlamorStyles, cssOverrides, cssProp],
+    [...styles, props.className, cssOverrides, cssProp],
```
Comment (kentcdodds) [inline src/get-glamor-classname.js]: niiiiiice

Comment (kentcdodds) [inline src/get-glamor-classname.js]: Actually, now that I think of it, this change probably would have worked before as well 😅

**Comment by kentcdodds** [comment]
Oh, and if you _could_ run your fancy benchmarks on this, I'd love to include it in my blogpost if there's any significant perf improvement here...

**Comment by tkh44** [comment]
I didn't see that much of an improvement in rerenders as I thought I would see.

Comparing the 2 fastest runs.

Before
```
- Rerender Duration: 206.05836666666664 ms
- Mount Duration: 180.5808 ms
```

After
```
- Rerender Duration: 196.3391 
- Mount Duration: 166.22899999999998 ms
```

That's about an 8% speedup in mount time. I would be careful to say that this is an 8% speed up though. Benchmarks are stupid 😄 This benchmark does not really test what this change is improving. It uses 1000 random numbers to generate elements with different styles on the page. You rarely hit the cache in that situation. On a real site with 100's of repeat classes and 1000's of elements, I could see this being a real perf win.

**Comment by kentcdodds** [comment]
That's great! I'll just say something like: "And @tkh44 may or may not have made glamorous 50x faster"

File: src/get-glamor-classname.js (hunk 4 above)
Comment (tkh44) [inline src/get-glamor-classname.js]: Done

**Pull request merged**